File: examples/typescript/basic_usage.ts
/**
 * MCP Discovery - TypeScript Example
 * Type-safe usage of the MCP Discovery API
 */

import { MCPDiscovery, MCPServer, DiscoveryResponse } from '@yksanjo/mcp-discovery';

async function main(): Promise<void> {
  // Create client with type safety
  const client = new MCPDiscovery();

  console.log('🔍 MCP Discovery - TypeScript Example\n');

  // Example 1: Basic discovery with full type safety
  console.log('Example 1: Find Slack integration');
  console.log('-'.repeat(40));

  const result: DiscoveryResponse = await client.discover('send slack notifications', { limit: 3 });

  console.log(`Query time: ${result.query_time_ms}ms`);
  console.log(`Total found: ${result.total_found}\n`);

  result.recommendations.forEach((server: MCPServer, i: number) => {
    console.log(`${i + 1}. ${server.name}`);
    console.log(`   Description: ${server.description}`);
    console.log(`   Install: ${server.install_command}`);
    console.log(`   Confidence: ${(server.confidence * 100).toFixed(0)}%`);
    console.log(`   Category: ${server.category}\n`);
  });

  // Example 2: Using with category filter
  console.log('\nExample 2: Filter by category');
  console.log('-'.repeat(40));

  const dbResult = await client.discover('database access', { 
    limit: 3, 
    category: 'database' 
  });

  console.log(`Found ${dbResult.total_found} database servers`);
  dbResult.recommendations.forEach((server: MCPServer) => {
    console.log(`• ${server.name}: ${server.install_command}`);
  });

  // Example 3: Get categories
  console.log('\n\nExample 3: Browse categories');
  console.log('-'.repeat(40));

  const categories = await client.getCategories();
  categories.slice(0, 5).forEach(cat => {
    console.log(`• ${cat.name} (${cat.count} servers)`);
  });

  // Example 4: Health check
  console.log('\n\nExample 4: Health check');
  console.log('-'.repeat(40));

  const isHealthy = await client.health();
  console.log(`API is ${isHealthy ? 'healthy ✅' : 'unhealthy ❌'}`);
}

// Run
main().catch(console.error);

File: examples/python/basic_usage.py
"""
MCP Discovery - Python Example
Basic usage of the MCP Discovery API
"""

import requests
from typing import List, Dict, Optional


class MCPDiscovery:
    """Python client for MCP Discovery API"""
    
    def __init__(self, api_url: str = "https://mcp-discovery-two.vercel.app"):
        self.api_url = api_url.rstrip('/')
    
    def discover(self, need: str, limit: int = 5) -> Dict:
        """
        Discover MCP servers matching a need.
        
        Args:
            need: Natural language description of what you need
            limit: Maximum number of results (1-50)
            
        Returns:
            Dictionary with recommendations, query_time_ms, total_found
        """
        response = requests.post(
            f"{self.api_url}/api/v1/discover",
            json={"need": need, "limit": limit}
        )
        response.raise_for_status()
        return response.json()
    
    def get_categories(self) -> List[Dict]:
        """Get all available categories."""
        response = requests.get(f"{self.api_url}/api/v1/categories")
        response.raise_for_status()
        return response.json()["categories"]


def main():
    # Create client
    client = MCPDiscovery()
    
    print("🔍 MCP Discovery - Python Example\n")
    
    # Example 1: Basic discovery
    print("Example 1: Find Slack integration")
    print("-" * 40)
    
    result = client.discover("send slack notifications", limit=3)
    
    print(f"Query time: {result['query_time_ms']}ms")
    print(f"Total found: {result['total_found']}")
    print()
    
    for i, server in enumerate(result['recommendations'], 1):
        print(f"{i}. {server['name']}")
        print(f"   Description: {server['description']}")
        print(f"   Install: {server['install_command']}")
        print(f"   Confidence: {server['confidence']:.0%}")
        print()
    
    # Example 2: Database search
    print("\nExample 2: Find database servers")
    print("-" * 40)
    
    result = client.discover("postgresql database", limit=3)
    
    for server in result['recommendations']:
        print(f"• {server['name']}: {server['install_command']}")
    
    # Example 3: List categories
    print("\n\nExample 3: Available categories")
    print("-" * 40)
    
    categories = client.get_categories()
    
    for category in categories[:5]:  # Show first 5
        print(f"• {category['name']}: {category['count']} servers")


if __name__ == "__main__":
    main()
